### PR TITLE
add map screen layout and import map api

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task xxxxx: Content](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
- add map screen layout
- import map api
- init fragment
- get current location
- handle back button, currrent location button
## Evidence (Screenshot or Video)
Screenshot_2024_0409_130943

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,6 +18,10 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.Androidtemplate"
         tools:targetApi="31">
+
+        <meta-data
+            android:name="com.google.android.geo.API_KEY"
+            android:value="hfkdshkfsdhjfhdsfksdhfjdsf"/>
```
Comment (daolq-2712) [inline app/src/main/AndroidManifest.xml]:
```suggestion
            android:value="hfkdshkfsdhjfhdsfksdhfjdsf"/>
```
Giấu key trong file giữ ở local

làm giống như ``app_name`` trong link này https://github.com/daolq-2712/Structure_Android/blob/mvp_kotlin/app/build.gradle.kts#L46

File: app/src/main/java/com/example/weather/screen/map/MapContract.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        fun onGetCurrentWeatherSuccess(weather: Weather)
+        fun onSearchLocation(latitude: Double, longitude: Double)
+        fun onInternetConnectionFailed()
+        fun onError(exception: Exception)
+    }
+
+    interface Presenter : BasePresenter<View> {
+        fun getWeather(latitude: Double, longitude: Double, isNetworkEnable: Boolean, isCurrent: Boolean)
+        fun insertWeather(weather: Weather)
+        fun deleteWeather(id: String)
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/screen/map/MapContract.kt]:
```suggestion
        fun favoriteWeather(weather: Weather)
        fun removeFavoriteWeather(id: String)
```

File: app/src/main/java/com/example/weather/screen/map/MapFragment.kt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    override fun initView() {
+        val mapFragment = childFragmentManager.findFragmentById(R.id.mapFragment) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+
+        viewBinding.currentLocationButton.setOnClickListener {
+            mGoogleMap?.clear()
+            zoomOnMap(LatLng(mCurrentLatitude, mCurrentLongitude))
+            addMarker(LatLng(mCurrentLatitude, mCurrentLongitude))
+        }
+        viewBinding.favoriteLocationButton.setOnClickListener { }
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/screen/map/MapFragment.kt]:
```suggestion
        viewBinding.favoriteLocationButton.setOnClickListener { 
        // TODO implement later
        }
```

File: app/src/main/java/com/example/weather/screen/map/MapFragment.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        viewBinding.icBack.setOnClickListener {
+            goBackFragment()
+        }
+        viewBinding.layoutWeatherMap.cardViewMap.setOnClickListener { }
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/screen/map/MapFragment.kt]: .

File: app/src/main/java/com/example/weather/screen/map/MapFragment.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        viewBinding.layoutWeatherMap.icStar.setOnClickListener { }
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/screen/map/MapFragment.kt]: .

File: app/src/main/java/com/example/weather/screen/map/MapFragment.kt [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                        Bitmap.createScaledBitmap(
+                            BitmapFactory.decodeResource(resources, R.drawable.ic_marker),
+                            Constant.MARKER_SIZE,
+                            Constant.MARKER_SIZE,
+                            false
+                        )
+                    )
+                )
+        ) // Custom Marker
+        return marker!!
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/screen/map/MapFragment.kt]: không force unwrap

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ghost) [reply app/src/main/AndroidManifest.xml]: dạ

File: app/src/main/java/com/example/weather/screen/map/MapPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,32 @@
+package com.example.weather.screen.map
+
+import com.example.weather.data.model.Weather
+import com.example.weather.data.repository.WeatherRepository
+
+class MapPresenter(
+    private val repository: WeatherRepository
+) : MapContract.Presenter {
+    override fun getWeather(latitude: Double, longitude: Double, isNetworkEnable: Boolean, isCurrent: Boolean) {
+        // todo later
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/screen/map/MapPresenter.kt]:
```suggestion
        // TODO implement later
```
mark để làm gì thì xem link sau sẽ rõ
https://stackoverflow.com/questions/23384804/how-to-see-all-todo-tasks-in-android-studio

File: app/src/main/res/layout/fragment_map.xml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        layout="@layout/layout_weather_map"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/dp_25"
+        android:layout_marginEnd="@dimen/dp_25"
+        android:layout_marginBottom="@dimen/dp_40"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:visibility="visible"/>
```
Comment (daolq-2712) [inline app/src/main/res/layout/fragment_map.xml]: default nó là visible rồi

File: app/src/main/res/layout/layout_weather_map.xml [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+            android:textSize="@dimen/sp_16"
+            android:fontFamily="@font/overpass_bold"
+            android:shadowColor="@color/textShadow"
+            android:shadowDx="2"
+            android:shadowDy="4"
+            android:shadowRadius="4"
+            android:layout_marginStart="@dimen/dp_25"
+            app:layout_constraintStart_toEndOf="@id/divide"
+            app:layout_constraintTop_toTopOf="parent"
+            android:text= "@string/h_ch_minh_vn" />
```
Comment (daolq-2712) [inline app/src/main/res/layout/layout_weather_map.xml]:
```suggestion
            android:text= "@string/h_ch_minh_vn" />
```
``h_ch_minh_vn`` không nên viết tắt như thế này

**Review by daolq-2712 — CHANGES_REQUESTED** [review]

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**